docs/_tutorials/read-only-json-api.md

```diff
@@ -0,0 +1,354 @@
+you need them, they behave the same way that posts do.
+
+Under `_posts`, create two new documents. Remember that Jekyll requires you to **prepend post filename with a date**:
+
+`2020–12–20-the-first-post.md`
+
+```md
+---
+title:  "The First Post"
+category: update
+custom-property: "My custom property"
+---
+1
+```
+
+Please note that you include a `custom-property` that you will fetch later on. 
+
+`2020–12–20-the-second-post.md`
+
+```md
+---
+title:  "The Second Post"
+categories: ["update", "tutorial"]
+---
+2
+```
+
+This one belongs to two categories --- `update` and `tutorial`, and doesn't include the custom property. Because the
+categories overlap, it will be possible to test getting a list of posts from a category.
+
+You'll also need a couple of layouts. Just add these empty files for now --- you'll fill (one of) them later:
+
+```
+.
+├── site
+│   ├── _layouts
+│   │   ├── category.html
+│   │   └── default.html
+```
+
+The last thing is to link the `default` layout in `config.yml`:
+
+```yml
+defaults:
+  -
+    scope:
+      path: ""
+    values:
+      layout: "default"
+```
+
+Since you won't have any pages in your project, leave the `path` to `""` in order to
+[cover]({% link _docs/configuration/front-matter-defaults.md %}) every file.
+
+## Convert Posts to JSON
+My approach is a little hacky --- it requires you to specify a JSON output format in a HTML template. The only thing
+you miss is that your text editor will most probably not recognize the markup file as a properly formatted JSON
+document. 
+
+### Define the Template
+Before you do it, it's worth noting that you can list all the available properties with this little template, which
+returns an array of available keys:
+
+{% raw %}
+```liquid
+{{ page.keys | jsonify }}
+```
+{% endraw %}
+
+
+If you enter it in a `default.html` template and execute the following command:
+
+```sh
+bundle exec jekyll build
+```
+
+You'll get this in `/_site/update/2020/12/20/the-first-post.html`:
+
+```json
+["url","relative_path","excerpt","title","categories","path","next",
+ "previous","id","output","date","tags","content","collection",
+ "draft","layout","category","custom-property","slug","ext"]
+```
+
+Given that, I'd like you to enter the following template instead of the aforementioned one-liner:
+
+{% raw %}
+```liquid
+{
+    "title": {{ page.title | jsonify }},
+    "categories": {{ page.categories | jsonify }},
+    "tags": {{ page.tags | jsonify }},
+    "content": {{ page.content | markdownify | jsonify  }},
+    "collection": {{ page.collection | jsonify }},
+    "date": {{ page.date | jsonify }},
+    "custom-property": {{ page.custom-property | jsonify }}
+}
+```
+{% endraw %}
+
+* Pass every property through a jsonify [filter]({% link _docs/liquid/filters.md %}#data-to-json). This will add
+quotation marks for us, convert arrays into strings and escape quotation marks.
```

Comment:
"This will add quotation marks for us, convert arrays into strings, and escape quotation marks."